test/Microsoft.Net.Http.Headers.Tests/HeaderUtilitiesTest.cs

```diff
@@ -210,5 +210,29 @@ public void TryParseNonNegativeInt32_Succeeds(string valueString, long expected)
             Assert.True(HeaderUtilities.TryParseNonNegativeInt32(valueString, out value));
             Assert.Equal(expected, value);
         }
+
+        [Theory]
+        [InlineData("\"hello\\\\\"", "hello\\")]
```

Comment:
Also test invalid quoted pairs e.g. `"\"hello\\\""`. Suspect that'll throw an `IndexOutOfRangeException` right now.